docs/bare_metal/c_small_libs.md

```diff
@@ -22,6 +22,29 @@ You can build with the smaller C libraries by creating an `mbed_app.json` with t
 
 This links your application with `microlib` for the `ARM` toolchain and `newlib-nano` for the `GCC_ARM` toolchain.
 
+### Non-returning main() required
```

Comment:
First question - if this is only for Arm microlib, why is it not in that section?

Reply:
The section here (how to enable small c_lib) is more noticeable. I was afraid if this is added to the Arm microlib section below, app developers may not always notice it.

Maybe I can move it to the microlib section and add a note here to point the user to have a look.